dev/src/bulk-writer.ts

```diff
@@ -728,3 +747,76 @@ export class BulkWriter {
     this.maxBatchSize = size;
   }
 }
+
+/**
+ * Validates the use of 'value' as BulkWriterOptions.
+ *
+ * @private
+ * @param arg The argument name or argument index (for varargs methods).
+ * @param value The object to validate.
+ * @throws if the input is not a valid BulkWriterOptions object.
+ */
+function validateBulkWriterOptions(arg: string | number, value: unknown): void {
+  if (validateOptional(value, {optional: true})) {
+    return;
+  }
+
+  if (!isObject(value)) {
+    throw new Error(
+      `${invalidArgumentMessage(
+        arg,
+        'bulkWriter() options argument'
+      )} Input is not an object.`
+    );
+  }
+
+  const options = value as firestore.BulkWriterOptions;
+
+  if (
+    options.throttling === undefined ||
+    typeof options.throttling === 'boolean'
+  ) {
+    return;
+  }
+
+  if ('initialOpsPerSecond' in options.throttling) {
+    validateInteger(
+      'initialOpsPerSecond',
+      options.throttling.initialOpsPerSecond,
+      {
+        minValue: 0,
+      }
+    );
+  }
+
+  if ('maxOpsPerSecond' in options.throttling) {
```

Comment:
If you change this to `options.throttling.maxOpsPerSecond !== undefined`, you can get rid of the bang below.